Content.Server/Weapons/Melee/EnergySword/EnergySwordSystem.cs

```diff
@@ -59,82 +61,102 @@ private void OnUseInHand(EntityUid uid, EnergySwordComponent comp, UseInHandEven
 
             if (comp.Activated)
             {
-                TurnOff(comp);
+                TurnOff(uid, comp);
             }
             else
             {
-                TurnOn(comp);
+                TurnOn(uid, comp);
             }
 
-            UpdateAppearance(comp);
+            UpdateAppearance(uid, comp);
         }
 
-        private void TurnOff(EnergySwordComponent comp)
+        private void TurnOff(EntityUid uid, EnergySwordComponent comp)
         {
             if (!comp.Activated)
                 return;
 
-            if (TryComp(comp.Owner, out ItemComponent? item))
+            if (TryComp(uid, out ItemComponent? item))
             {
-                _item.SetSize(comp.Owner, 5, item);
+                _item.SetSize(uid, 5, item);
             }
 
-            if (TryComp<DisarmMalusComponent>(comp.Owner, out var malus))
+            if (TryComp<DisarmMalusComponent>(uid, out var malus))
             {
-                malus.Malus -= comp.litDisarmMalus;
+                malus.Malus -= comp.LitDisarmMalus;
             }
 
-            if(TryComp<MeleeWeaponComponent>(comp.Owner, out var weaponComp))
+            if(TryComp<MeleeWeaponComponent>(uid, out var weaponComp))
             {
                 weaponComp.HitSound = comp.OnHitOff;
                 if (comp.Secret)
                     weaponComp.HideFromExamine = true;
             }
 
             if (comp.IsSharp)
-                RemComp<SharpComponent>(comp.Owner);
+                RemComp<SharpComponent>(uid);
 
-            _audio.Play(comp.DeActivateSound, Filter.Pvs(comp.Owner, entityManager: EntityManager), comp.Owner, true, comp.DeActivateSound.Params);
+            if (TryComp<ReflectProjectileComponent>(uid, out var reflectProj))
+            {
+                reflectProj.Enabled = false;
+            }
+
+            if (TryComp<ReflectHitScanComponent>(uid, out var reflectHitScan))
+            {
+                reflectHitScan.Enabled = false;
+            }
```

Comment:
I know the rest of this is shitcode but please don't add it to it, just use an event sub on another system.

Reply:
Ok, i found many cases where item can be activated: energy sword, stun baton, lights, pinpointer and many more i think. What can be done is to add a component, that will have a single field `bool Activated` and 2 events: deactivated event and activated event. I don't want to refactor all of it in this PR tho.